### PR TITLE
shared/rng: Rename sol_rng_engine to sol_random

### PR DESCRIPTION
**Pull request by lpereira**
This renames `sol_rnd_engine` to `sol_random` so it's easier to figure
out what it does from a cursory glance.

Also slightly change the API to use a `sol_buffer` instead of
writing bytes to an `unsigned char` array.  While at it, add a few
utility functions to generate integers, doubles, and booleans, without
requiring the user to actually instantiate a `sol_buffer`.

The internal API also changed slightly: each implementation now
generates an `uint32_t`, which is appended to the buffer given to
`sol_random_fill_buffer()` function.

### CODE REVIEW

**Comment by lpereira** [comment]
Ping @barbieri, @glima.

**Comment by lpereira** [comment]
(The force-push was to fix a style issue and keep Semaphore happy.)

**Comment by lpereira** [comment]
(Another `push -f` to rebase/fix conflicts.)

File: src/lib/datatypes/sol-buffer.c [hunk 1]
```diff
@@ -38,18 +38,25 @@
...
-    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_FIXED_CAPACITY, -EPERM);
-    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_NO_FREE, -EPERM);
 
     if (buf->capacity == new_size)
         return 0;
 
+    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_FIXED_CAPACITY, -EPERM);
+    SOL_EXP_CHECK(buf->flags & SOL_BUFFER_FLAGS_NO_FREE, -EPERM);
+
     new_data = realloc(buf->data, new_size);
```
Comment (barbieri) [inline src/lib/datatypes/sol-buffer.c]: why did you move these checks? I wanted to make sure nobody calls resize on not-owned buffers.

wanted to know what would break with that :-)

Comment (lpereira) [inline src/lib/datatypes/sol-buffer.c]: These checks were moved, not removed. Shouldn't make a difference if old_size == new_size. (Maybe the move isn't necessary anymore, will check before pushing.)

Comment (barbieri) [inline src/lib/datatypes/sol-buffer.c]: i said they were moved... that's why I'm wondering. Nobody should call resize on fixed buffers. If the capacity == new_size, it will not warn... and will make spotting call sites harder (ie: I test on my cases where this is equal and no warnings, then you use in your app and suddenly get some warnings to that same code that got in unoticed).

Comment (lpereira) [inline src/lib/datatypes/sol-buffer.c]: Moving those checks is not required anymore, they were just artifacts.

**Comment by lpereira** [comment]
If there are no other comments, I'll merge this soon.

**Comment by lpereira** [comment]
Merged.
